pkg/network/network.go

```diff
@@ -93,49 +94,131 @@ func IsChangeSafe(prev, next *opv1alpha1.NetworkAddonsConfigSpec) error {
 	return nil
 }
 
+// Render creates component object list
 func Render(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, openshiftNetworkConfig *osv1.Network, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
 	log.Print("starting render phase")
 	objs := []*unstructured.Unstructured{}
-
-	// render Multus
-	o, err := renderMultus(conf, manifestDir, openshiftNetworkConfig, clusterInfo)
-	if err != nil {
-		return nil, err
+	o := []*unstructured.Unstructured{}
+	var err error
+
+	if conf.KubeMacPool != nil {
```

Comment:
Why are you changing this function? Please rever the changes if not needed.

Reply:
we need it, it was in the function , but sometimes the function (rendernmstate) can allow nil so moved it from there
(on removal it allow conf nil if prev != nil && prev.nmstate != nil)

reorg is possible but then the render itself will need to get prev as arg as well

Reply:
Are you talking about your new render2remove or the original one?

Reply:
the original one, but there the if was at renderNMState itself,
but i removed it from renderNMState , because now i need two different conditions on the callers of renderNMState
one conf != nil ,and the other (upon destroy) the long cond above

Reply:
i have an idea maybe, 
i will put the long condition
conf.module != nil || (prev != nil && prev.module != nil)
as an acceptance gate into each of the moduleRender functions

and then on the RenderDelete i will give conf and prev,
but on Render i will give just conf, so the condition would be as it was before,
depend just on conf

this way we can also toggle out new feature in order to test the failing tests
but if i am not wrong, indeed to reorder there can do troubles
we will get to it

Reply:
i am doing the suggested change above just to the kmp, so you can view and say if its ok

btw after that (or maybe not as a part of this PR), we can take it to a next level of creating an array of objects and iterate on it, each with its own fields, but first let fix all the test and such, this is more advance go practice i guess

Reply:
added a commit that show a suggested change only to kmp , so you see if you like the approach please